backend/TraderX/src/main/java/cmpe451/group6/rest/transaction/service/TransactionService.java

```diff
@@ -0,0 +1,204 @@
+import cmpe451.group6.rest.asset.model.Asset;
+import cmpe451.group6.rest.asset.repository.AssetRepository;
+import cmpe451.group6.rest.equipment.configuration.EquipmentConfig;
+import cmpe451.group6.rest.equipment.model.Equipment;
+import cmpe451.group6.rest.equipment.repository.EquipmentRepository;
+import cmpe451.group6.rest.transaction.dto.TransactionDTO;
+import cmpe451.group6.rest.transaction.model.Transaction;
+import cmpe451.group6.rest.transaction.model.TransactionType;
+import cmpe451.group6.rest.transaction.repository.TransactionRepository;
+import cmpe451.group6.authorization.model.User;
+import cmpe451.group6.authorization.repository.UserRepository;
+import cmpe451.group6.rest.follow.model.FollowStatus;
+import cmpe451.group6.rest.follow.repository.FollowRepository;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Date;
+
+@Service
+public class TransactionService {
+
+    @Autowired
+    private TransactionRepository TransactionRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private FollowRepository followRepository;
+
+    @Autowired
+    private EquipmentRepository equipmentRepository;
+
+    @Autowired
+    private AssetRepository assetRepository;
+
+    @Autowired
+    private ModelMapper modelMapper;
+
+    public List<Transaction> getTransactionsByUser(String username, String requesterName) {
+        User user = userRepository.findByUsername(username);
+        if (user == null) {
+            throw new CustomException("There is no user named " + username + ".", HttpStatus.NOT_ACCEPTABLE);
+        } else if (user.getIsPrivate() && (followRepository.getFollowDAO(requesterName, username) == null
+                || followRepository.getFollowDAO(requesterName, username).getFollowStatus() != FollowStatus.APPROVED)) {
+            throw new CustomException("The requested user's profile is private and requester is not following!",
+                    HttpStatus.NOT_ACCEPTABLE);
+        } else {
+            List<Transaction> transactions = TransactionRepository.findByUser_username(username);
+            return transactions;
+        }
+    }
+
+    public List<TransactionDTO> getTransactions() {
+        List<TransactionDTO> transactionDTOs = new ArrayList<TransactionDTO>();
+        TransactionRepository.findAll()
+                .forEach(item -> transactionDTOs.add(modelMapper.map(item, TransactionDTO.class)));
+        return transactionDTOs;
+    }
+
+    public List<TransactionDTO> getTransactionsByCode(String code) {
+        List<TransactionDTO> transactionDTOs = new ArrayList<TransactionDTO>();
+        TransactionRepository.findByEquipment_code(code)
+                .forEach(item -> transactionDTOs.add(modelMapper.map(item, TransactionDTO.class)));
```

Comment:
We have to consider request sender and private profile issues.

Reply:
See comment https://github.com/bounswe/bounswe2019group6/pull/178#discussion_r347967844